modules/runner/src/integrationTest/java/org/apache/ignite/internal/index/ItIndexManagerTest.java

```diff
@@ -71,7 +71,7 @@ public void eventsAreFiredWhenIndexesCreatedAndDropped() {
         CompletableFuture<IndexEventParameters> indexCreatedFuture = registerListener(indexManager, IndexEvent.CREATE);
 
         indexManager.createIndexAsync("PUBLIC", "INAME", "TNAME", true, tableIndexChange ->
-                tableIndexChange.convert(HashIndexChange.class).changeColumnNames("C3", "C2")).join();
+                tableIndexChange.convert(HashIndexChange.class).changeColumnNames("C3", "C2")).get();
```

Comment:
you keep potentially infinity lock.
Let's use `await(CompletionStage stage)` instead of any 'join', `get`

Reply:
The main issue here was the join() is not interruptible, but get() is.

Reply:
Agree, but with `get` we could wait for a test timeout instead of a short wait timeout in the utility method.

Reply:
+1 for @AMashenkov approach - we need not guessing each test work time.